scripts/exglobal_land_analysis_run.sh

```diff
@@ -0,0 +1,175 @@
+#! /usr/bin/env bash
```

Comment:
I don't understand what is going in this script, but seems like we are deviating from the `execute` method from `LandAnalysis`.
@CoryMartin-NOAA 
Do you have an understanding of the requirements for the `execute`?

Reply:
This script needs to be reworked significantly.
Is it possible to outline what is being done in this script in a flow diagram so we can suggest a solution?

Reply:
Yes, you are right. This script includes some from `initialize`, all the `execute`, and some from `finalize`. There are four separate sections in this run script `scripts/exglobal_land_analysis_run.sh` for the snow DA. They are 1) create the background ensembles; 2) process the observed IMS data; 3) conduct JEDI runs; and 4) apply the JEDI increments. We have discussed and agreed to use the shell scripts here for this snow DA implementation initially due to the limited time. 

The above four processes were implemented in the `GDASApp` separately as below: 

1. Create the background ensembles ([Issue #199](https://github.com/NOAA-EMC/GDASApp/issues/199) and [PR #207](https://github.com/NOAA-EMC/GDASApp/pull/207)): A python code is used to convert the model restart files into two separate ensembles. 

2. IMS preprocessing ([Issue #198](https://github.com/NOAA-EMC/GDASApp/issues/198) and [PR #216](https://github.com/NOAA-EMC/GDASApp/pull/216)): Code for processing IMS input ascii/netcdf files on the UFS model grid. 

- Inputs: IMS ascii/netcdf file, IMS index file (is model resolution specific) and model restart files. 
- Output: file with i) IMS-derived snow cover fraction over land on UFS model grid, and ii) snow depth derived form the IMS snow cover fraction, using an inversion of the noah model snow depletion curve. 

3. Conduct JEDI run ([Issue #241](https://github.com/NOAA-EMC/GDASApp/issues/241) and [PR #248](https://github.com/NOAA-EMC/GDASApp/pull/248)): Scripts to perform the snow DA, to assimilate GTS snow depth and IMS snow cover obs, using the JEDI LETKF and for the noahMP land surface model.

4. Apply the DA increments generated by JEDI to UFS sfc_data restart ([Issue #200](https://github.com/NOAA-EMC/GDASApp/issues/200) and [PR #233](https://github.com/NOAA-EMC/GDASApp/pull/233)). Currently, only option is to add snow depth increment to the Noah-MP land surface model. Above based on disaggregation code written by Mike Barlage.

Reply:
@jiaruidong2017, @aerorahul and I spoke today and we think this is probably the eventual path forward:
1 and 3 will go into the landanlrun job
2 will get a new landprep job (for eventual transition to obsproc)
4 will go to landanlfinal job
Does that make sense/sound reasonable?

Reply:
@CoryMartin-NOAA Can we talk how to proceed with this? Thanks.

Reply:
I have added a new `landanlprep` job for the IMS preprocessing. The following codes were added: 
```
        jobs/JGLOBAL_LAND_ANALYSIS_PREP
        jobs/rocoto/landanlprep.sh
        parm/config/config.landanlprep
        scripts/exglobal_land_analysis_prep.sh
```

@CoryMartin-NOAA For `4 will go to landanlfinal job` you mentioned above, I will need help on how to move this part (a shell script) into the `landanlfinal` job (a python code).  Thanks.